src/validation/tests/array_validation_test.rs

```diff
@@ -346,3 +346,52 @@ fn array_assignment_function_call() {
 
     assert_snapshot!(diagnostics);
 }
+
+#[test]
+fn validate_ranges() {
+    let diagnostics = parse_and_validate_buffered(
+        r"
+        FUNCTION foo
+            VAR
+                arr_A : ARRAY[1..-5] OF DINT;
+                arr_B : ARRAY[1..5] OF ARRAY[1..-10] OF DINT;
+                arr_C : ARRAY[1..-5] OF ARRAY[1..-10] OF DINT;
+                arr_D : ARRAY[1..5, 1..-5] OF DINT;
+            END_VAR
+        END_FUNCTION
+        ",
+    );
+
+    assert_snapshot!(diagnostics, @r###"
+    error[E097]: Invalid range `1..-5`, did you mean `-5..1`?
+      ┌─ <internal>:4:17
+      │
+    4 │                 arr_A : ARRAY[1..-5] OF DINT;
+      │                 ^^^^^ Invalid range `1..-5`, did you mean `-5..1`?
```

Comment:
Do we store the source location of array ranges somewhere? I wasn't able to find anything.

Reply:
The array will initially be created as a DatatypeDefinition and the index visitor will make it a reference. The Definition still however hold the original information. Which includes the original AST Datatype containing an Array type with a `bounds` field which is the `AstNode` for the array bounds. See `plc_ast/src/ast.rs` `DataType#ArrayType` and  `DataTypeDeclaration`. The problem I see is that this information is not transfered to the DatatypeInformation once we index. It might make sense to extend the Dimension to include the AstNode that created it (In another commit)